File: LCA-I/Python Implementation/LCATest.py
import unittest
import LCA
from Node import Node

class testLCA(unittest.TestCase):

    def testContains(self):

        # Create tree
        root = Node(1)
        root.left = Node(2)
        root.left.left = Node(3)
        root.left.right = Node(4)
        root.right = Node(5)
        root.right.left = Node(6)
        root.right.right = Node(7)

        self.assertTrue(LCA.contains(root, Node(3)), "Check if tree with root provided contains a specified node")
        self.assertFalse(LCA.contains(root, Node(9)), "Check contains() with node not in tree")

    def testgetLCA(self):

        # Create tree
        root = Node(1)
        root.left = Node(2)
        root.left.left = Node(3)
        root.left.right = Node(4)
        root.right = Node(5)
        root.right.left = Node(6)
        root.right.right = Node(7)

        self.assertEqual(LCA.getLCA(root, Node(3), Node(4)).id, 2, "Check LCA of two nodes contained in tree")
        self.assertEqual(LCA.getLCA(root, Node(7), Node(2)).id, 1, "Check getLCA() with nodes of different depths")
        self.assertIsNone(LCA.getLCA(root, Node(0), Node(4)), "Check getLCA() when one node is not on tree")
        


if __name__ == '__main__':
    unittest.main()

File: LCA-I/Java Implementation/LCATest.java
import static org.junit.Assert.*;

import org.junit.Test;

public class LCATest {
	
	public Node root;
	
	//For simplisity sake for this particular program
	public void createTree() {
		root = new Node(1);
		root.left = new Node(2);
		root.left.left = new Node(3);
		root.left.right = new Node(4); 
		root.right = new Node(5);
		root.right.left = new Node(6);
		root.right.right = new Node(7);
	} 

	@Test
	public void testContains() {
		
		createTree();		
		
		assertTrue("Check if contains() can navigate left and right to a leaf node", LCA.contains(root, new Node(4)));
		assertFalse("Check node not contained in supplied tree", LCA.contains(root, new Node(8))); 
	}
	
	//Tests both getLCA and getLCAr
	@Test
	public void testGetLCA() {

		createTree();
		
		assertEquals("Check LCA where one node is the root", 1, LCA.getLCA(root, new Node(1), new Node(1)).id);
		assertEquals("Check LCA of two nodes contained on tree where LCA is not the root", 2, LCA.getLCA(root, new Node(3), new Node(4)).id); 
		assertEquals("Check LCA of two nodes contained on tree at different depths", 1, LCA.getLCA(root, new Node(3), new Node(7)).id);
		assertNull("Check LCA of when first node isn't contained on the tree", LCA.getLCA(root, new Node(9), new Node(7))); 
		assertNull("Check LCA of when second node isn't contained on the tree", LCA.getLCA(root, new Node(1), new Node(9))); //For the sake of full code coverage
		
	}

}
